### PR TITLE
Update the maintainers guide to include a candidate check

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: dev/MAINTAINING.md [hunk 1]
```diff
@@ -138,8 +138,12 @@
...
           files lying around in the workspace and unintentionally ship
           them as well.
-          * `cabal v2-sdist`; `cabal upload --publish <tarpath>`
-          * `stack upload .`
+          * Upload and verify the candidate
+            * `cabal v2-sdist; cabal upload <tarpath>`
+            * `cabal v2-haddock --haddock-for-hackage --enable-doc; cabal upload -d <tarpath>`
+          * Publish the package
+            * `cabal v2-sdist`; `cabal upload --publish <tarpath>`
+        * `stack upload .`
```
Comment (harendra-kumar) [inline dev/MAINTAINING.md:146]: This is not equivalent to the cabal workflow, so let's remove it.

Comment (adithyaov) [reply dev/MAINTAINING.md:146]: Where do we move it to?
This is on the same level as `Upload to hackage`. It isn't coming under the cabal workflow.

Comment (harendra-kumar) [reply dev/MAINTAINING.md:146]: Remove it.

Comment (harendra-kumar) [reply dev/MAINTAINING.md:146]: Delete this line.

**Pull request merged**